cmd/influxd/inspect/report_tsm/report_tsm.go

```diff
@@ -0,0 +1,356 @@
+	_, _ = fmt.Fprintln(tw, strings.Join([]string{"DB", "RP", "Shard", "File", "Series", "New" + estTitle, "Min Time", "Max Time", "Load Time"}, "\t"))
+
+	minTime, maxTime := int64(math.MaxInt64), int64(math.MinInt64)
+	var fileCount int
+	if err := a.walkShardDirs(a.dir, func(db, rp, id, path string) error {
+		if a.pattern != "" && strings.Contains(path, a.pattern) {
+			return nil
+		}
+
+		file, err := os.OpenFile(path, os.O_RDONLY, 0600)
+		if err != nil {
+			_, _ = fmt.Fprintf(cmd.ErrOrStderr(), "error: %s: %v. Skipping.\n", path, err)
+			return nil
+		}
+
+		loadStart := time.Now()
+		reader, err := tsm1.NewTSMReader(file)
+		if err != nil {
+			_, _ = fmt.Fprintf(cmd.ErrOrStderr(), "error: %s: %v. Skipping.\n", file.Name(), err)
+			return nil
+		}
+		loadTime := time.Since(loadStart)
+		fileCount++
+
+		dbCount := dbCardinalities[db]
+		if dbCount == nil {
+			dbCount = newCounterFn()
+			dbCardinalities[db] = dbCount
+		}
+
+		oldCount := dbCount.Count()
+
+		seriesCount := reader.KeyCount()
+		for i := 0; i < seriesCount; i++ {
+			key, _ := reader.KeyAt(i)
+			totalSeries.Add(key)
+			dbCount.Add(key)
+
+			if a.detailed {
+				sep := strings.Index(string(key), "#!~#")
+				seriesKey, field := key[:sep], key[sep+4:]
+				measurement, tags := models.ParseKey(seriesKey)
+
+				measCount := measCardinalities[measurement]
+				if measCount == nil {
+					measCount = newCounterFn()
+					measCardinalities[measurement] = measCount
+				}
+				measCount.Add(key)
+
+				fieldCount := fieldCardinalities[measurement]
+				if fieldCount == nil {
+					fieldCount = newCounterFn()
+					fieldCardinalities[measurement] = fieldCount
+				}
+				fieldCount.Add(field)
+
+				for _, t := range tags {
+					tagCount := tagCardinalities[string(t.Key)]
+					if tagCount == nil {
+						tagCount = newCounterFn()
+						tagCardinalities[string(t.Key)] = tagCount
+					}
+					tagCount.Add(t.Value)
+				}
+			}
+		}
+		minT, maxT := reader.TimeRange()
+		if minT < minTime {
+			minTime = minT
+		}
+		if maxT > maxTime {
+			maxTime = maxT
+		}
+		_ = reader.Close()
+
+		_, _ = fmt.Fprintln(tw, strings.Join([]string{
+			db, rp, id,
+			filepath.Base(file.Name()),
+			strconv.FormatInt(int64(seriesCount), 10),
+			strconv.FormatInt(int64(dbCount.Count()-oldCount), 10),
+			time.Unix(0, minT).UTC().Format(time.RFC3339Nano),
+			time.Unix(0, maxT).UTC().Format(time.RFC3339Nano),
+			loadTime.String(),
+		}, "\t"))
+		if a.detailed {
+			_ = tw.Flush()
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	_ = tw.Flush()
+
+	printArgs{
+		fileCount:          fileCount,
+		minTime:            minTime,
+		maxTime:            maxTime,
+		estTitle:           estTitle,
+		totalSeries:        totalSeries,
+		detailed:           a.detailed,
+		tagCardinalities:   tagCardinalities,
+		measCardinalities:  measCardinalities,
+		fieldCardinalities: fieldCardinalities,
+		dbCardinalities:    dbCardinalities,
+	}.printSummary(cmd)
```

Comment:
I think this would be cleaner if `printSummary` took `printArgs` as an input instead of making it a method